CMake use mono (USDU-160)

judubu opened the pull request:
## Purpose of this PR

**Ticket/Jira #:** https://jira.unity3d.com/browse/USDU-160

USD.NET can now be built on OSX and CMake looks for a Mono install as opposed to a Unity.
Fixes some gen type issue which was messing up bindings signatures tfor int64/uint64 types.
Bindings tests are now back as a CMake component but won't work until the USD.NET.Unity move PR lands (it fixes the tests by removing the Unity dependency)

## Testing

**Functional Testing status:**

The package loads fine and all tests pass

**Performance Testing status:**
N/A

## Overall Product Risks
<!-- See Testing Status, Complexity and Halo Effect for your PR](https://confluence.unity3d.com/display/DEV/Testing+Status%2C+Complexity+and+Halo+Effect+for+your+PR). -->

**Complexity:**
<!-- (Minimal / Low / Medium / High) -->
Low

**Halo Effect:**
<!-- (Minimal / Low / Medium / High) -->
Low

## Additional information

**Note to reviewers:**

<!-- Info per person for what to focus on, or historical info to understand who have previously reviewed and coverage. Help them get context. -->

judubu committed:
CMake now search for Mono instead of Unity

Also fixes a bunch of build warnings for duplicated files
Also fixes the Mono_PInvoke decoration script to prevent decorating multiple times.

judubu committed:
Update build script shebang and don't install USD.NET.dll.mdb

judubu committed:
Update binaries

judubu committed:
Enable the tests component in CMake.
Will work fine when the USD.NET.Unity move branch is merged

judubu committed:
Fix vt type info generation script for messed up int64 and uint64

clusty reviewed (approved):
Looks good (gibberish :) ):

judubu committed:
Fix USD.NET build for osx with custom Mono.

Building USD.NET with custom Mono is now optional. Default is still to use the version provided by Unity.

judubu committed:
Update Unity Hub OSX path for CMake

judubu committed:
Fix typo in build script
